### PR TITLE
CI fixes and other stuff

### PR DESCRIPTION
**Pull request by miguelalonsojr**
### Proposed change(s)

Nit to core package, update package manifest, update build provider to address this bug: https://unity.slack.com/archives/C8FECS6L9/p1669646048318809

### Types of change(s)

- [X] Bug fix
- [ ] New feature
- [ ] Code refactor
- [ ] Breaking change
- [ ] Documentation update
- [ ] Other (please describe)

### Checklist
- [ ] Added tests that prove my fix is effective or that my feature works
- [ ] Updated the changelog (if applicable)
- [ ] Updated the documentation (if applicable)
- [ ] Updated the migration guide (if applicable)

### CODE REVIEW

File: com.unity.ml-agents/Editor/MLAgentsSettingsBuildProvider.cs [hunk 1]
```diff
@@ -9,61 +9,29 @@ namespace Unity.MLAgents.Editor
...
-            m_SettingsAddedToPreloadedAssets = null;
+            if (!EditorUtility.IsPersistent(MLAgentsSettingsManager.Settings))
+                return;
 
             var preloadedAssets = PlayerSettings.GetPreloadedAssets().ToList();
             if (!preloadedAssets.Contains(MLAgentsSettingsManager.Settings))
             {
-                m_SettingsAddedToPreloadedAssets = MLAgentsSettingsManager.Settings;
-                preloadedAssets.Add(m_SettingsAddedToPreloadedAssets);
+                preloadedAssets.Add(MLAgentsSettingsManager.Settings);
```
Comment (taozhuo) [inline com.unity.ml-agents/Editor/MLAgentsSettingsBuildProvider.cs:22]: is `MLAgentsSettingsManager.Settings` guaranteed to be non-null? I guess so, I found this in a test file:

`Assert.AreNotEqual(null, MLAgentsSettingsManager.Settings);`

but not 100% sure

File: com.unity.ml-agents/Editor/MLAgentsSettingsBuildProvider.cs [hunk 2]
```diff
@@ -9,61 +9,29 @@ namespace Unity.MLAgents.Editor
...
         public void OnPostprocessBuild(BuildReport report)
         {
-            if (m_SettingsAddedToPreloadedAssets == null)
-                return;
-
-            var wasDirty = IsPlayerSettingsDirty();
-
             var preloadedAssets = PlayerSettings.GetPreloadedAssets().ToList();
-            if (preloadedAssets.Contains(m_SettingsAddedToPreloadedAssets))
+            if (preloadedAssets.Contains(MLAgentsSettingsManager.Settings))
```
Comment (taozhuo) [inline com.unity.ml-agents/Editor/MLAgentsSettingsBuildProvider.cs:30]: Same as above.

**Review by taozhuo — APPROVED** [review]

**Review by taozhuo — APPROVED** [review]

**Comment by taozhuo** [comment]
LGTM

File: com.unity.ml-agents/Editor/MLAgentsSettingsBuildProvider.cs (hunk 1 above)
Comment (miguelalonsojr) [reply com.unity.ml-agents/Editor/MLAgentsSettingsBuildProvider.cs:22]: Yes. See https://github.com/Unity-Technologies/ml-agents/blob/6bb711f1b0a29a06b4deb544bb1e93c392acb255/com.unity.ml-agents/Runtime/MLAgentsSettingsManager.cs#L22.

File: com.unity.ml-agents/Editor/MLAgentsSettingsBuildProvider.cs (hunk 2 above)
Comment (miguelalonsojr) [reply com.unity.ml-agents/Editor/MLAgentsSettingsBuildProvider.cs:30]: See above response.

**Review by taozhuo — APPROVED** [review]

**Pull request merged**